File: web_version/main/PasswordGenerator/components/link_button.py
import reflex as rx


def link_button(icon, text: str, url: str, style, size="", is_external=True) -> rx.Component:
    return rx.link(
        rx.button(icon, text, size=size,),
        href=url,
        is_external=True if is_external else False,
        style=style,
        
    )

File: v_web_app/frontend/src/app/components/LinkButton.js
import Link from 'next/link';
import {Button} from "@nextui-org/button";

export default function LinkButton({ icon, text, url, style, size = "", isExternal = true}) {
    return (
        <Link target={isExternal ? "_blank" : "_self"} rel={isExternal ? "noopener noreferrer" : undefined} href={url} passHref style={style}>
            <Button
                className={'flex justify-between items-center p-2 rounded-lg transition-colors duration-200 bg-slate-300 text-zinc-900 hover:text-slate-100 hover:bg-zinc-900'}
                style={{ fontSize: size }}
            >
                {icon} {text}
            </Button>
        </Link>
    );
}